src/qiskit_toqm/toqm_plugin.py

```diff
@@ -0,0 +1,62 @@
+from qiskit.transpiler.preset_passmanagers.plugin import PassManagerStagePlugin
+from qiskit.transpiler.preset_passmanagers import common
+from qiskit_toqm import *
+
+
+class ToqmSwapPlugin(PassManagerStagePlugin):
+
+    def pass_manager(self, pass_manager_config, optimization_level):
+        opt_level_to_strategy = {
+            0: ToqmStrategyO0,
+            1: ToqmStrategyO1,
+            2: ToqmStrategyO2,
+            3: ToqmStrategyO3,
+        }
+
+        toqm_strategy_preset = opt_level_to_strategy.get(optimization_level, ToqmStrategyO3)
```

Comment:
Heh, I like default to the most optimization :)

Reply:
I don't remember the exact motivation for why I made `optimization_level` an optional field in the plugin interface. I think there was a test case early in the development of the feature which ended up not sending the level, but that shouldn't be the case anymore. I guess it's still a possibility if someone calls the plugin directly outside of the preset passmanager interface.